virtualcluster/pkg/syncer/vnode/provider/provider.go

```diff
@@ -18,10 +18,34 @@ package provider
 
 import (
 	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/constants"
+	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/util/featuregate"
+	utilconstants "sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/util/constants"
 )
 
 // VirtualNodeProvider is the interface used for registering the node address.
 type VirtualNodeProvider interface {
 	GetNodeDaemonEndpoints(node *corev1.Node) (corev1.NodeDaemonEndpoints, error)
 	GetNodeAddress(node *corev1.Node) ([]corev1.NodeAddress, error)
+	GetLabelsToSync() map[string]struct{}
+}
+
+// GetNodeLabels is used to sync allowed node labels to vNode
+func GetNodeLabels(p VirtualNodeProvider, node *corev1.Node) map[string]string {
+	labels := map[string]string{
+		constants.LabelVirtualNode: "true",
+	}
+
+	if featuregate.DefaultFeatureGate.Enabled(featuregate.SuperClusterPooling) {
+		labels[constants.LabelSuperClusterID] = utilconstants.SuperClusterID
+	}
+
+	labelsToSync := p.GetLabelsToSync()
+	for k, v := range node.GetLabels() {
+		if _, found := labelsToSync[k]; found {
+			labels[k] = v
+		}
+	}
```

Comment:
Does this need to go through the same filtering we do on https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/virtualcluster/pkg/syncer/conversion/equality.go#L117-L162 so that we don't sync labels labels in the VirtualCluster CR?

Reply:
The list is pretty straightforward as comes from the command line flags, so the operator should explicitly add the label to be synced.
On the other hand, the CheckDWObjectMetaEquality method is not used for vNode and pNode comparison, so there are no conflicts.